ykman/cli/otp.py

```diff
@@ -313,13 +317,47 @@ def yubiotp(ctx, slot, public_id, private_id, key, no_enter, force,
             key = click.prompt('Enter secret key', err=True)
             key = a2b_hex(key)
 
+    if upload:
+        click.echo('Will upload secret key to YubiCloud.')
+
     force or click.confirm('Program an OTP credential in slot {}?'.format(slot),
                            abort=True, err=True)
+
+    if upload:
+        upload_result = controller.prepare_upload_key(
+            key, public_id, private_id)
+
     try:
-        controller.program_otp(slot, key, public_id, private_id, not no_enter)
+        if (not upload) or (upload and upload_result['success']):
+            controller.program_otp(
+                slot, key, public_id, private_id, not no_enter)
     except YkpersError as e:
         _failed_to_write_msg(ctx, e)
 
+    if upload:
+        if upload_result['success']:
+            click.echo('Upload to YubiCloud initiated successfully. '
+                       'Please finish the upload procedure in your browser.')
+            webbrowser.open_new_tab(upload_result['url'])
+            click.echo('If the browser did not open automatically, '
```

Comment:
I think the two messages could be shortened to one, `Upload initiated. Opening [URL] in browser.` or similar